unyt/array.py

```diff
@@ -2280,3 +2281,37 @@ def savetxt(
         newline="\n",
         comments=comments,
     )
+
+
+def allclose_units(actual, desired, rtol=1e-7, atol=0, **kwargs):
+    # Create a copy to ensure this function does not alter input arrays
+    act = unyt_array(actual)
+    des = unyt_array(desired)
+
+    try:
+        des = des.in_units(act.units)
+    except (UnitOperationError, UnitConversionError):
+        return False
+
+    rt = unyt_array(rtol)
+    if not rt.units.is_dimensionless:
+        raise RuntimeError("Units of rtol (%s) are not " "dimensionless" % rt.units)
+
+    if not isinstance(atol, unyt_array):
+        at = unyt_quantity(atol, des.units)
+    else:
+        at = atol
+
+    try:
+        at = at.in_units(act.units)
+    except UnitOperationError:
+        return False
```

Comment:
codecov reports that this line is still untested

Reply:
Yes, I saw that and I was wondering how to hit that exception..

Reply:
Something like this, I think:

```
with pytest.raises(UnitOperationError):
    allclose_units([1, 2, 3]*kg, [1, 2, 3]*kg, atol=3*meter)
```

(haven't actually tried running that, please double-check me)